react/components/SearchResultContainer.js

```diff
@@ -265,9 +259,9 @@ SearchResult.propTypes = SearchResultWithData.propTypes = {
   /** Maximum number of items per page. */
   maxItemsPerPage: PropTypes.number.isRequired,
   /** Query param. e.g: eletronics/smartphones */
-  query: PropTypes.string,
+  query: PropTypes.string.isRequired,
```

Comment:
I think we have a common denominator, a bunch of props that can be reused.

Reply:
What props?

Reply:
In SearchResult.js ```
query: PropTypes.shape({
+      /**
+       * Rest of the search term, e.g: eletronics/smartphones/samsung implies that
+       * rest will be equal to "smartphones,samsung".
+       * */
+      rest: PropTypes.string,
+      /** Determines the types of the terms, e.g: "c,c,b" (category, category, brand). */
+      map: PropTypes.string,
+      /** Search's pagination.  */
+      page: PropTypes.string,
+      /** Search's ordenation. */
+      order: PropTypes.string,
+    }),
```